### PR TITLE
🚸 Display latest events before fetching all events  

### PR DESCRIPTION
**Pull request by WeiJun0827**

### CODE REVIEW

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/mixins/nft.js [hunk 1]
```diff
@@ -517,109 +515,95 @@ export default {
...
-          }
-        });
-      }
-      history.sort((a, b) => b.timestamp - a.timestamp);
+            // add original event after grant event for time reverse order
+            historyWithGrant.push(event);
+          });
+          this.NFTHistory.push(...historyWithGrant);
+        } else {
+          this.NFTHistory.push(...history);
```
Comment (williamchong) [inline src/mixins/nft.js:571]: When do we reset `this.NFTHistory`?

Comment (WeiJun0827) [reply src/mixins/nft.js:571]: Probably safer to reset it before pushing. Nice caught.

Comment (williamchong) [reply src/mixins/nft.js:571]: Would be easier to read if you return `...historyWithGrant` or `...history` in this function
then explicitly call push when calling i.e. `this.NFTHistory.push(addGrantEventForWNFTAndUpdateUserInfo(...))`

Comment (williamchong) [reply src/mixins/nft.js:571]: Would be even cleaner if we completely avoid using `this`, and use function param instead in `addGrantEventForWNFTAndUpdateUserInfo`

Comment (WeiJun0827) [reply src/mixins/nft.js]: updated, see if it's cleaner

**Review by williamchong — APPROVED** [review]

**Review by nwingt — APPROVED** [review]

**Pull request merged**